engine/src/main/java/org/hibernate/search/bridge/util/impl/NumericFieldUtils.java

```diff
@@ -118,9 +118,12 @@ public static boolean requiresNumericRangeQuery(Object value) {
 			return false;
 		}
 		return value instanceof Double ||
+				value instanceof Byte ||
+				value instanceof Short ||
 				value instanceof Long ||
 				value instanceof Integer ||
 				value instanceof Float ||
+				value instanceof Date ||
```

Comment:
Regarding `Date` and `Calendar`, we'd have to take the backend into account as well, in order to a account for a different mapping in ES. Should we track this as a separate issue?